src/App.js

```diff
@@ -1,25 +1,64 @@
 import React from 'react';
 
 import './App.scss';
+import { Clock } from './components/Clock';
 
-const App = () => {
-  setInterval(() => {
-    const date = new Date();
+export class App extends React.Component {
+  state = {
+    isClockVisible: true,
+    clockName: Math.random(),
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.clockName === this.state.clockName) {
+      return;
+    }
 
     // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
+    console.log(
+      'The Clock was renamed from '
+      + `${prevState.clockName} to ${this.state.clockName}`,
+    );
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+  render() {
+    return (
+      <div className="App column is-one-third container">
+        <div className="panel">
+          <div className="panel-block">
+            <button
+              type="button"
+              onClick={
+                () => this.setState({ isClockVisible: true })
+              }
+              className="button is-small is-light is-fullwidth mr-3"
+            >
+              Show Clock
+            </button>
+            <button
+              type="button"
+              onClick={
+                () => this.setState({ isClockVisible: false })
+              }
+              className="button is-small is-light is-fullwidth mr-3"
+            >
+              Hide Clock
+            </button>
+            <button
+              type="button"
+              onClick={
+                () => this.setState({ clockName: Math.random() })
+              }
```

Comment:
а если код функции нужно расширить?
```suggestion
              onClick={() => {
                this.setState({ clockName: Math.random() })
              }}
```